### PR TITLE
Removes excess helper functions

### PR DESCRIPTION
**Pull request by krhoda**
Increases nonce to 11 chars to exceed 64 bits of entropy

Updates API to be more idiomatic

Updated README to match

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,13 +124,18 @@ iex> Siwe.validate(parsed, String.trim(sig))
...
+- `parse_if_valid_sig` takes a SIWE message string and `signature`, then returns a parsed `Message` only if the `signature` matches.
 
-- `parse_if_valid_time!` takes a SIWE message string then returns a parsed `Message` only if the current time is after the `Message`'s `not_before` field (if it exists) and before the `Message`'s `expiration_time` field (if it exists). 
+- `parse_if_valid` takes a SIWE message string and a `signature` then returns a parsed `Message` only if the `signature` matches and the current time is after the `Message`'s `not_before` field (if it exists) and before the `Message`'s `expiration_time` field (if it exists). 
 
-- `parse_if_valid!` takes a SIWE message string and a `signature` then returns a parsed `Message` only if the `signature` matches and the current time is after the `Message`'s `not_before` field (if it exists) and before the `Message`'s `expiration_time` field (if it exists). 
+Another helper, `generate_nonce` function is provided to create alphanumeric [a-z, A-Z, 1-9] nonces compliant with SIWE's spec, used like so:
+This is useful for servers using SIWE for their own authentication systems.
```
Comment (obstropolos) [inline README.md:134]: Move this line below example code

**Review by wyc — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -96,12 +97,13 @@ This library deals with three different types of input:
   end
 ```
 
-The most basic functions are `from_str!` and `to_str!` which translate a SIWE message string to a parsed SIWE message and back (respectively). To simplify using the variables from the above example:
+The most basic functions are `parse` and `to_str` which translate a SIWE message string to a parsed SIWE message and back (respectively). To simplify using the variables from the above example:
 
 ```
-iex> parsed = Siwe.from_str!(String.trim(msg))
+iex> {:ok, parsed} = Siwe.from_str!(String.trim(msg))
```
Comment (G-0-0) [inline README.md:103]:
```suggestion
iex> {:ok, parsed} = Siwe.parse!(String.trim(msg))
```

File: README.md [hunk 3]
```diff
@@ -122,14 +124,19 @@ iex> Siwe.validate(parsed, String.trim(sig))
 :true
 ```
 
-A trio of optimized helpers can be used to combine these steps:
+A pair of optimized helpers can be used to combine these steps:
 
-- `parse_if_valid_sig!` takes a SIWE message string and `signature`, then returns a parsed `Message` only if the `signature` matches.
+- `parse_if_valid_sig` takes a SIWE message string and `signature`, then returns a parsed `Message` only if the `signature` matches.
```
Comment (G-0-0) [inline README.md:129]: Seems that `parse_if_valid_sig` has been removed too?

**Review by G-0-0 — COMMENTED** [review]
There's just a couple of small mistakes in the README I think.

File: native/siwe_ex/src/lib.rs [hunk 4]
```diff
@@ -75,17 +79,17 @@ fn message_to_parsed(m: Message) -> Parsed {
...
     ))
 }
 
 #[rustler::nif]
-fn to_str(message: Parsed) -> NifResult<String> {
+fn to_str(message: Parsed) -> Result<String, String> {
     Ok(message
         .to_eip4361_message()
-        .map_err(|_e| Error::BadArg)?
+        .map_err(|e| format!("Faile to marshal to string: {}", e))?
```
Comment (G-0-0) [inline native/siwe_ex/src/lib.rs:92]: Oh sorry just noticed a small typo here too: `Faile` -> `Failed`

File: README.md (hunk 2 above)
Comment (krhoda) [reply README.md:103]: Fixed this, along with a few other typos.

File: README.md (hunk 3 above)
Comment (krhoda) [reply README.md:129]: Removed references to that

File: native/siwe_ex/src/lib.rs (hunk 4 above)
Comment (krhoda) [reply native/siwe_ex/src/lib.rs]: Thank you! Wouldn't have caught that.

**Pull request merged**